.github/workflows/update-rpc.yml

```diff
@@ -0,0 +1,94 @@
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: kakarot-rpc
+
+      - name: Check Current Submodule Commit
+        id: check_submodule_commit
+        run: |
+          cd kakarot-rpc/lib/kakarot
+          current_commit=$(git rev-parse HEAD)
+          latest_commit=$(git ls-remote https://github.com/kkrt-labs/kakarot.git ${{ env.release_tag }} | awk '{print $1}')
+          echo "current_commit=${current_commit}" >> $GITHUB_ENV
+          echo "latest_commit=${latest_commit}" >> $GITHUB_ENV
+          cd -
+
+      - name: Check if Branch Exists
+        id: check_branch_exists
+        run: |
+          cd kakarot-rpc
+          if git ls-remote --exit-code --heads origin kakarot-${{ env.latest_commit }}; then
+            echo "branch_exists=true" >> $GITHUB_ENV
+          else
+            echo "branch_exists=false" >> $GITHUB_ENV
+          fi
+          cd -
+
+      - name: Debug Print Environment Variables
+        run: |
+          echo "Current commit: ${{ env.current_commit }}"
+          echo "Latest commit: ${{ env.latest_commit }}"
+          echo "Branch exists: ${{ env.branch_exists }}"
+
+      - name: Update Submodule
+        if: env.current_commit != env.latest_commit && env.branch_exists == 'false'
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+          cd kakarot-rpc/lib/kakarot
+          echo "Current directory: $(pwd)"
+          git fetch --all
+          git checkout ${{ env.latest_commit }}
+          cd -
+          cd kakarot-rpc
+          echo "Current directory: $(pwd)"
+          git checkout -b kakarot-${{ env.latest_commit }}
+          git add lib/kakarot
+          git commit -m "Update kakarot to ${{ env.release_tag }}"
+          git push --set-upstream origin kakarot-${{ env.latest_commit }}
+          cd -
+
+      - name: Check if Pull Request Exists
+        id: check_pr_exists
+        run: |
+          pr_exists=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                        -H "Accept: application/vnd.github+json" \
+                        "https://api.github.com/repos/kkrt-labs/kakarot-rpc/pulls?state=open" \
+                        | jq '[.[] | select(.head.ref == "kakarot-${{ env.latest_commit }}")] | length > 0')
+          echo "pr_exists=${pr_exists}" >> $GITHUB_ENV
+
+      - name: Create Pull Request
+        if: env.current_commit != env.latest_commit && env.pr_exists == 'false'
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            https://api.github.com/repos/kkrt-labs/kakarot-rpc/pulls \
+            -d '{"title":"Update kakarot submodule to ${{ env.release_tag }}","head":"kakarot-${{ env.latest_commit }}","base":"main","body":"This PR updates the kakarot submodule to the latest release ${{ env.release_tag }}."}'
```

Comment:
what about https://stackoverflow.com/a/73340290/4444546 ?

```
  steps:
    - name: create pull request
      run: gh pr create -B base_branch -H branch_to_merge --title 'Merge branch_to_merge into base_branch' --body 'Created by Github action'
      env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Nice just modified, should be equivalent but nicer